tensorboard/plugins/graph/tf_graph_board/tf-graph-board.ts

```diff
@@ -309,4 +310,29 @@ class TfGraphBoard extends LegacyElementMixin(PolymerElement) {
       ? node.include
       : tf_graph.InclusionType.UNSPECIFIED;
   }
+  @observe('graph')
+  _slimGraphChanged() {
+    // By default, avoid coloring by 'structure' for large graphs, since it may be very expensive.
+    // Color by 'structure' is still available to users via explicit gesture.
+    if (!this.graph) {
+      return;
+    }
+    const isGraphTooLarge =
+      Object.keys(this.graph.nodes).length > 10000 &&
+      this.graph.edges.length > 10000;
+    if (isGraphTooLarge && this.colorBy === ColorBy.STRUCTURE) {
+      this.colorBy = ColorBy.NONE;
+      this.fire('color-by-changed', this.colorBy);
```

Comment:
AFAIK, literally a setter on `this.colorBy` triggers this exact event. Was it not the case?

Reply:
Ah yes Polymer magic.  Setting `this.colorBy` only triggers this event when the property has `notify: true`.  Since `.fire()` is part of the LegacyElementMixin, I've dropped it in favor of notify: true.